Add-ons/UmbracoForms/Developer/IPreValueTextFileStorage/index.md

```diff
@@ -0,0 +1,74 @@
+---
+versionFrom: 9.0.0
+versionTo: 10.0.0
+---
+
+# Storing prevalue text files with IPreValueTextFileStorage
+
+Umbraco Forms contains a built-in `Get value from textfile` [Prevalue Source Type](../Extending/Adding-a-Prevaluesourcetype.md) that stores the uploaded text file into the physical file system (by default in  `umbraco\Data\UmbracoForms\PreValueTextFiles`).
+
+You can replace the default implementation by writing your own `IPreValueTextFileStorage` and registering that using e.g. `builder.Services.AddUnique<IPreValueTextFileStorage, CustomPreValueTextFileStorage>()` (in `Startup.cs` or a composer).
+
+Since Forms 10.2, you can also use/inherit from `PreValueTextFileSystemStorage` to change the underlying `IFileSystem` that's used to store the prevalue text files.
+
+## Move files to Media file system (10.2 and up)
```

Comment:
Besides 10.2, we're also planning to backport this to 9.6 and 8.14. The setup code for v9 will be the same, but for v8 the composer and service registration will be slightly different. What's the best way to add these samples, create separate pages per version?

Reply:
The convention is to have an `index-v8.md` file with appropriate meta data at the top:

```
---
versionFrom: 8.0.0
versionTo: 8.0.0
meta.Title: "Content Apps"
---
```

See `\Add-ons\UmbracoForms\Developer\ContentApps\` for an example.

Reply:
This can be done in a separate PR as it'll be available from when the final release goes out, rather than when the RC of 10.2 is.